src/systemd.c

```diff
@@ -281,3 +283,50 @@ int cgroup_create_scope(const char * const scope_name, const char * const slice_
 
 	return ret;
 }
+
+int cgroup_create_scope2(struct cgroup *cgroup, int ignore_ownership,
+			 const struct cgroup_systemd_scope_opts * const opts)
+{
+	char *copy1 = NULL, *copy2 = NULL, *slice_name, *scope_name;
+	int error = 0;
+
+	if (!cgroup)
+		return ECGROUPNOTALLOWED;
+
+	copy1 = strdup(cgroup->name);
+	if (!copy1) {
+		error = ECGOTHER;
+		goto err;
+	}
+
+	scope_name = basename(copy1);
+
+	copy2 = strdup(cgroup->name);
+	if (!copy2) {
+		error = ECGOTHER;
+		goto err;
+	}
+
+	slice_name = dirname(copy2);
+
+	error = cgroup_create_scope(scope_name, slice_name, opts);
+	if (error)
+		goto err;
+
+	/*
+	 * cgroup_create_cgroup() can gracefully handle EEXIST if the cgroup already exists, so
+	 * let's just call it because it already knows how to manage the ownership of the cgroup
+	 * hierarchy.
+	 */
+	error = cgroup_create_cgroup(cgroup, ignore_ownership);
+	if (error)
+		goto err;
```

Comment:
@drakenclimber Thanks for working on the idea, it looks awesome.  Can you please add more comments on why we need to call `cgroup_create_cgroup()`?

Reply:
I guess the check for `error` can be removed too.

Reply:
Good idea on both accounts.